src/mmap.rs

```diff
@@ -448,10 +448,18 @@ impl GuestMemory for GuestMemoryMmap {
     }
 
     fn find_region(&self, addr: GuestAddress) -> Option<&GuestRegionMmap> {
-        for region in self.regions.iter() {
-            if addr >= region.start_addr() && addr <= region.end_addr() {
-                return Some(region);
+        let bs_region = self.regions.binary_search_by_key(&addr, |x| x.start_addr());
```

Comment:
just a style proposal: I would merge this line into the following `match` by writing `match self.regions.binary_search_by_key(&addr, |x| x.start_addr()) { ... }`

Reply:
Taking into account all of your suggestion, I rewrote everything as a match expression, at the cost of a slightly more complicated `if` clause.